### PR TITLE
Comment float test cases and add test cases for interpreter

### LINKED ISSUE #3: Test interpreter for false positives
Currently, there are only tests for the positive outcomes of the interpreter.  Negative/error cases should be tested too, similar to what we have in parser tests. No need to check exact error messages as those are not very useful yet anyway, just need to make sure it fails where it should.

Examples of failing programs:
```
1 / 0 # can't divide by zero
```
```
a + 1 # no variable 'a'
```
```
# Condition should be boolean
if "something" then
  1
else
  0
end
```

There's plenty of those, ideally, we need to cover all cases where `Left` is returned from `evaluate` function in `Eval.sh`

If you find a case where it should fail but it doesn't, please create an issue.

### PR DESCRIPTION
**Pull request by mateidanut**
The operations on floating point numbers are not yet implemented, but these test cases contain such operations.
I commented them so that the test suite now passes completely.

Also, I added tests for invalid programs that will pass the Parser stage but not the Interpreter one.

This PR closes #3

### CODE REVIEW

File: test/Interpreter/EvalSpec.hs [hunk 1]
```diff
@@ -275,3 +289,161 @@ spec = do
     |]
 
     testProgramm map (List [Integer 2, Integer 3, Integer 4])
+
+    -- test for calling a non-existant function
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs]: I'd say `test for trying to call non-function or getting a length of non-list` or something similar.

File: test/Interpreter/EvalSpec.hs [hunk 2]
```diff
@@ -275,3 +289,161 @@ spec = do
...
+    let badCall = [r|
+        let f = 7
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs]: Use two spaces as a tab to be consistent with the rest of the code

File: test/Interpreter/EvalSpec.hs [hunk 3]
```diff
@@ -275,3 +289,161 @@ spec = do
...
+    |]
+
+    testIncorrectProgramm invalidConcat
+
+    -- test for algebraic op between non-numbers
+    let badAlgebraic = [r|
+        let f = 41
+        let g = True
+        f - g
+    |]
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs:477]: Could you add other algebraic operators too, please?

File: test/Interpreter/EvalSpec.hs [hunk 4]
```diff
@@ -275,3 +289,161 @@ spec = do
...
+
+    testIncorrectProgramm badAlgebraic
+
+    -- test for boolean op between non-booleans
+    let badBooleanOp = [r|
+        let f = False
+        let g = "string"
+        f && g
+    |]
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs:486]: Could you add other boolean operators too, please?

File: test/Interpreter/EvalSpec.hs [hunk 5]
```diff
@@ -275,3 +289,161 @@ spec = do
...
+
+    testIncorrectProgramm badBooleanOp
+
+    -- test for comparison between non-integers
+    let badComp = [r|
+        let f = "string1"
+        let g = "string2"
+        f < g
+    |]
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs:495]: Same as above, but for comparison operators :)

**Comment by mateidanut** [comment]
Done :)

**Comment by yurtsiv** [comment]
@mateidanut One minor thing I noticed is that `testProgramm` needs to be renamed to `testCorrectProgramm`.  Just for consistency. Thanks :)

**Comment by mateidanut** [comment]
Sure thing, I renamed it

**Comment by yurtsiv** [comment]
@mateidanut Thanks. Please, fix the conflict and I'll merge

**Comment by yurtsiv** [comment]
Hah, pushed to your fork instead of the original repo :)

**Comment by mateidanut** [comment]
It happens sometimes :)

**Comment by yurtsiv** [comment]
@mateidanut Are you finished with renaming? Currently, there's an error when running the tests

**Comment by mateidanut** [comment]
@yurtsiv Sorry for the delay, there was one more line that I hadn't payed attention to when renaming. It should work properly now

**Comment by yurtsiv** [comment]
@mateidanut Great, thanks!

**Pull request merged**